app/controllers/documents_controller.rb

```diff
@@ -0,0 +1,66 @@
+class DocumentsController < ApplicationController
+    before_action :authenticate_user!, except: [:public_show]
+
+  def index
+    @documents = current_user.documents
+    @documents_count = @documents.count
+  end
+
+  def new
+    @document = current_user.documents.new
+  end
+
+  def create
+    @document = current_user.documents.new(document_params)
+    if @document.save
```

Comment:
Can @document.save raise exception?
Please handle this

Reply:
added exception